[18.0][MIG] account_banking_ach_direct_debit: Migration to 18.0

xaviedoanhduy opened the pull request:
- `allow_blocked` was remove from this commit https://github.com/OCA/bank-payment/commit/43c010f
- since both `allow_blocked (account.payment.line.create)` and `blocked (account.move.line)` no longer exist, the domain of the `move_line_domain` field has been changed, causing more valid records to be returned, see: https://github.com/OCA/bank-payment/commit/43c010f#diff-40552b
- since `account.payment` is no longer inherit on `account.move` but is now optional in this commit https://github.com/odoo/odoo/pull/180813/commits/2609fc2e38, the `mandate_id` field will be checked against the `move_id` of each payment.
- when I click the **Generate Payment File** button, an error comes from another module, see: `https://github.com/OCA/l10n-usa/pull/146#issuecomment-2890252177`

Commit:
l10n_us_form_1099

oca-transbot committed:
OCA Transbot updated translations from Transifex

max3903 committed:
l10n_us_form_1099: Migration to 11.0

JordiBForgeFlow committed:
add l10n_us_gaap and l10n_us_mis_report

JordiBForgeFlow committed:
add l10n_us_gaap and l10n_us_mis_report

nikul-serpentcs committed:
[MIG] Migrated l10n_us_form_1099 v11 to v12

oca-travis committed:
[UPD] Update l10n_us_form_1099.pot

dduarte-odoogap committed:
[MIG] l10n_us_gaap: Migration to 12.0

max3903 committed:
[ADD] account_banking_ach_direct_debit

murtuzasaleh committed:
[IMP] account_banking_ach_direct_debit

oca-travis committed:
[UPD] Update l10n_us_form_1099.pot

Commit:
[ADD] readme folder and add keys in manifest file

[IMP] adding first unit tests

oca-travis committed:
[UPD] Update l10n_us_gaap_mis_report.pot

oca-travis committed:
[UPD] Update account_banking_ach_direct_debit.pot

OCA-git-bot committed:
[ADD] icon.png

brian10048 committed:
[IMP] l10us_form_1099

Commit:
[ADD] Following points:
- Add a new object 'type.1099' with a char field 'Name',
- Add a many2one field '1099 Type' on res.partner to type.1099,
- Add data for type.1099 with: 1099-A, 1099-B, 1099-C, 1099-CAP, 1099-DIV, 1099-G, 1099-H, 1099-INT, 1099-K, 1099-LTC, 1099-MISC, 1099-OID, 1099-PATR, 1099-Q, 1099-R, 1099-S, 1099-SA, RRB-1099 and SSA-1099
- Add security for new object

Commit:
[FIX] CI, make unique ID for access right

brian10048 committed:
[IMP] start of 1099 report

Commit:
[FIX] travis and CI, change Char to Many2one type_1099 field in SQL report

brian10048 committed:
[FIX] data and manifest

Remove noupdate for 1099 data
Shorten summary in manifest

brian10048 committed:
[IMP] review comments

oca-travis committed:
[UPD] Update l10n_us_form_1099.pot

OCA-git-bot committed:
l10n_us_form_1099 12.0.1.1.0

max3903 committed:
[IMP] l10n_us_form_1099: Add 1099-MISC boxes

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-usa-12.0/l10n-usa-12.0-l10n_us_form_1099
Translate-URL: https://translation.odoo-community.org/projects/l10n-usa-12-0/l10n-usa-12-0-l10n_us_form_1099/

oca-travis committed:
[UPD] Update l10n_us_form_1099.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
l10n_us_form_1099 12.0.1.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-usa-12.0/l10n-usa-12.0-l10n_us_form_1099
Translate-URL: https://translation.odoo-community.org/projects/l10n-usa-12-0/l10n-usa-12-0-l10n_us_form_1099/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update l10n_us_gaap.pot

OCA-git-bot committed:
[UPD] README.rst

MiquelRForgeFlow committed:
[MIG] l10n_us_gaap_mis_report: Migration to 12.0

murtuzasaleh committed:
[13.0][MIG] l10n_us_form_1099

OCA-git-bot committed:
[UPD] README.rst

JoanSForgeFlow committed:
[IMP] l10n_us_gaap: black, isort

JoanSForgeFlow committed:
[MIG] l10n_us_gaap : Migration to 13.0

JoanSForgeFlow committed:
[IMP] l10n_us_gaap_mis_report: black, isort

JoanSForgeFlow committed:
[MIG] l10n_us_gaap_mis_report: Migration to 13.0

AdriaGForgeFlow committed:
[ADD] stock interim and price differences accounts

AdriaGForgeFlow committed:
[ADD] Inventory Adjustment account

oca-travis committed:
[UPD] Update l10n_us_gaap.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
pre-commit update

oca-travis committed:
[UPD] Update l10n_us_gaap_mis_report.pot

OCA-git-bot committed:
[UPD] README.rst

JordiBForgeFlow committed:
l10n_us_gaap should use anglo saxon setting

OCA-git-bot committed:
l10n_us_gaap 13.0.1.1.0

JordiBForgeFlow committed:
Added translation using Weblate (Spanish)

JordiBForgeFlow committed:
Added translation using Weblate (Spanish)

JordiBForgeFlow committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (124 of 124 strings)

Translation: l10n-usa-13.0/l10n-usa-13.0-l10n_us_gaap_mis_report
Translate-URL: https://translation.odoo-community.org/projects/l10n-usa-13-0/l10n-usa-13-0-l10n_us_gaap_mis_report/es/

JevinD committed:
[IMP] l10n_us_form_1099: black, isort, prettier

JevinD committed:
[MIG] l10n_us_form_1099: migration to 14.0

ran precommit after making edits

[IMP] fix Read me Usage workflow

[IMP]fixed newline at end of file

JevinD committed:
[FIX] l10n_us_form_1099: change value of key in manifest

Chanakya-SerpentCS committed:
[FIX] Travis

Chanakya-SerpentCS committed:
[13.0][MIG]account_banking_ach_direct_debit

oca-travis committed:
[UPD] Update l10n_us_form_1099.pot

OCA-git-bot committed:
[UPD] README.rst

Murtaza-OSI committed:
[14.0][MIG] account_banking_ach_direct_debit Module.

Murtaza-OSI committed:
[IMP] Code after pre-commit.

Freni-OSI committed:
[ADD] Tests

oca-travis committed:
[UPD] Update account_banking_ach_direct_debit.pot

OCA-git-bot committed:
[UPD] README.rst

JordiBForgeFlow committed:
[l10n_us_gaap][mig] migration to 14.0

JordiBForgeFlow committed:
[l10n_us_gaap] fix tests

ChrisOForgeFlow committed:
[15.0][MIG][l10n_us_gaap, l10n_us_gaap_mis_report]

oca-travis committed:
[UPD] Update l10n_us_gaap.pot

OCA-git-bot committed:
[UPD] README.rst

lk-eska committed:
l10n_us_gaap: Migration to 16.0

lk-eska committed:
[MIG] l10n_us_form_1099: Migration to 16.0

Murtaza-OSI committed:
[16.0][MIG] account_banking_ach_direct_debit

[FIX] account_banking_ach_direct_debit: use the payment_ids field instead of bank_line_ids

Apply same fix as #101

JordiBForgeFlow committed:
[IMP] l10n_us_gaap_mis_report: Update code prefixes

JordiBForgeFlow committed:
[MIG] l10n_us_gaap_mis_report - Migration to 16.0

Commit:
[UPD] Update l10n_us_form_1099.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-usa-16.0/l10n-usa-16.0-l10n_us_form_1099
Translate-URL: https://translation.odoo-community.org/projects/l10n-usa-16-0/l10n-usa-16-0-l10n_us_form_1099/

Commit:
[UPD] Update l10n_us_gaap.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-usa-16.0/l10n-usa-16.0-l10n_us_gaap
Translate-URL: https://translation.odoo-community.org/projects/l10n-usa-16-0/l10n-usa-16-0-l10n_us_gaap/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

dreispt committed:
[REF] l10n_us_gaap: pre-comit fixes

Commit:
[UPD] Update l10n_us_gaap_mis_report.pot

OCA-git-bot committed:
[BOT] post-merge updates

AnaisLForgeFlow committed:
[IMP] l10n_us_form_1099: pre-commit auto fixes

AnaisLForgeFlow committed:
[MIG] l10n_us_form_1099: Migration to 17.0

Commit:
[UPD] Update l10n_us_form_1099.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-usa-17.0/l10n-usa-17.0-l10n_us_form_1099
Translate-URL: https://translation.odoo-community.org/projects/l10n-usa-17-0/l10n-usa-17-0-l10n_us_form_1099/

JordiBForgeFlow committed:
[UPD] change version to 16.0

JordiBForgeFlow committed:
[FIX] account_banking_ach_direct_debit: fix pylint errors

Commit:
[ADD] l10n_us_mis_financial_report

Commit:
[UPD] Update l10n_us_mis_financial_report.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[FIX] l10n_us_mis_financial_report: wrong tag in Balance Domain

OCA-git-bot committed:
[BOT] post-merge updates

JasminSForgeFlow committed:
[IMP] l10n_us_gaap: black, isort, prettier

JasminSForgeFlow committed:
[MIG] l10n_us_gaap: Migration to 17.0

JasminSForgeFlow committed:
[IMP] l10n_us_gaap_mis_report: black, isort, prettier

JasminSForgeFlow committed:
[MIG] l10n_us_gaap_mis_report: Migration to 17.0

Commit:
[IMP] l10n_us_mis_financial_report: pre-commit auto fixes

Commit:
[MIG] l10n_us_mis_financial_report: Migration to 17.0

Commit:
[UPD] Update l10n_us_mis_financial_report.pot

OCA-git-bot committed:
[BOT] post-merge updates

JasminSForgeFlow committed:
[IMP] l10n_us_gaap: black, isort, prettier

JasminSForgeFlow committed:
[MIG] l10n_us_gaap: Migration to 18.0

JasminSForgeFlow committed:
[IMP] l10n_us_gaap_mis_report: black, isort, prettier

JasminSForgeFlow committed:
[MIG] l10n_us_gaap_mis_report: Migration to 18.0

sakthi-shan committed:
[MIG] l10n_us_form_1099: Migration to 18.0

xaviedoanhduy committed:
[IMP] account_banking_ach_direct_debit: pre-commit auto fixes

xaviedoanhduy committed:
[MIG] account_banking_ach_direct_debit: Migration to 18.0

JordiMForgeFlow committed:
[IMP] account_banking_ach_base: remove duplicate code from l10n_us_account_routing

nguyenminhchien commented on account_banking_ach_direct_debit/README.rst:
```diff
@@ -0,0 +1,111 @@
+For defining a payment mode that uses ACH direct debit:
+
+1. Go to *Accounting > Configuration > Management > Payment Modes*.
+2. Create a Customer Invoice.
```
i think this step should be: `Create a new Payment Mode`

xaviedoanhduy replied:
I think you are misunderstanding between Payment Mode and Payment Order because this module has imported 1 ACH payment mode in `data.xml` file, so we just need to create invoices using that available Payment Mode instead of creating new ones

and  maybe the author is directing users to create `Debit Order` -> `Payment Order` from 1 invoice instead of creating each manually separately

nguyenminhchien commented on account_banking_ach_direct_debit/README.rst:
```diff
@@ -0,0 +1,111 @@
+Usage
+=====
+
+In the menu *Accounting > Payments > Debit Order*, create a new debit
```
```suggestion
In the menu *Accounting > Customers > Debit Order*, create a new debit
```

nguyenminhchien commented on account_banking_ach_direct_debit/models/account_payment_order.py:
```diff
@@ -0,0 +1,65 @@
+        for order in self:
+            to_expire_mandates = first_mandates = all_mandates = Mandate
+            for payment in order.payment_ids.filtered("move_id.mandate_id"):
+                mandate_id = payment.move_id.mandate_id
```
```suggestion
                mandate = payment.move_id.mandate_id
```
It's a recordset, `mandate` is better for naming convention

nguyenminhchien replied:
Take the mandate from payment.order, that field is added by `https://github.com/OCA/bank-payment/pull/1470`

xaviedoanhduy replied:
> Take the mandate from payment.order, that field is added by https://github.com/OCA/bank-payment/pull/1470


cool, it was merged after we investigated

nguyenminhchien reviewed (changes requested):
I faced this error when generating the payment file. It's caused by `account_banking_ach_base` (not your module). But, i can not go further.
```
  File "/home/chien/code/odoo/odoo/master/odoo/http.py", line 727, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/home/chien/code/odoo/odoo/master/addons/web/controllers/dataset.py", line 42, in call_button
    action = call_kw(request.env[model], method, args, kwargs)
  File "/home/chien/code/odoo/odoo/master/odoo/api.py", line 533, in call_kw
    result = getattr(recs, name)(*args, **kwargs)
  File "/home/chien/code/oca/bank-payment/18.0/account_payment_order/models/account_payment_order.py", line 386, in open2generated
    payment_file_str, filename = self.generate_payment_file()
  File "/home/chien/code/oca/l10n-usa/18.0/account_banking_ach_direct_debit/models/account_payment_order.py", line 19, in generate_payment_file
    return self.generate_ach_file()
  File "/home/chien/code/oca/l10n-usa/18.0/account_banking_ach_base/models/account_payment_order.py", line 135, in generate_ach_file
    self.validate_mandates(line)
  File "/home/chien/code/oca/l10n-usa/18.0/account_banking_ach_base/models/account_payment_order.py", line 75, in validate_mandates
    if not line.mandate_id:
AttributeError: 'account.payment' object has no attribute 'mandate_id'
```

nguyenminhchien reviewed (commented):
I faced this error when clicking on `Generate Payment File`  button, it caused by `account_banking_ach_base` (not your module). But, failed at this step, i cannot move to next step to check the button `File Successfully Uploaded`
Do you face the same issue? 

![image](https://github.com/user-attachments/assets/7d6854b6-f71c-4a01-858f-7335e96656e7)

xaviedoanhduy commented:
> I faced this error when clicking on `Generate Payment File` button, it caused by `account_banking_ach_base` (not your module). But, failed at this step, i cannot move to next step to check the button `File Successfully Uploaded` Do you face the same issue?
> 
> ![image](https://private-user-images.githubusercontent.com/8724793/446543070-7d6854b6-f71c-4a01-858f-7335e96656e7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDc5MTM0NTYsIm5iZiI6MTc0NzkxMzE1NiwicGF0aCI6Ii84NzI0NzkzLzQ0NjU0MzA3MC03ZDY4NTRiNi1mNzFjLTRhMDEtODU4Zi03MzM1ZTk2NjU2ZTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MjJUMTEyNTU2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjZiNGE5MGVmZDliNzY0MmNhY2FmN2IxZDY1MDQxN2M3MzM0NGNlOGM3Y2RjY2MwOTM1ODFkNjE0MTY4ZmZmNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ECLvZnLwpowiK36s7dP2u3r007ehWgPgZd5cx_WnwWc)

xaviedoanhduy commented:
> Do you face the same issue?

You must make sure that some values ​​like `routing_number`, `legal_id_number` and `acc_number` are all numeric strings, for example `015545665` in both Account Journal and Partner on payment line.

https://github.com/xaviedoanhduy/l10n-usa/blob/71da02216474a6c87975e637403b268c90ac4f6a/account_banking_ach_base/models/account_payment_order.py#L52-L58

it should be like this https://pypi.org/project/ach/:
```
settings = {
    'immediate_dest' : '123456789', # Your bank's routing number
    'immediate_org' : '123456789', # Bank assigned routing number
    'immediate_dest_name' : 'YOUR BANK',
    'immediate_org_name' : 'YOUR COMPANY',
    'company_id' : '1234567890', #tax number
}
```

If successful the result will look like this: [2025-05-22_3_A.txt](https://github.com/user-attachments/files/20391138/2025-05-22_3_A.txt)

xaviedoanhduy commented:
I was going to write a function to validate these data lines. But then, I realized this error has appeared before and each country will have a more appropriate account number so it will be difficult to have this error in a production environment.

amh-mw committed:
[ADD] l10n_us_base_county, l10n_us_crm_county

Import counties from U.S. Census Bureau data [^1] [^2] and add county field to partners and leads.

[^1]: https://www.census.gov/library/publications/2011/compendia/usa-counties-2011.html
[^2]: https://www2.census.gov/library/publications/2011/compendia/usa-counties/excel/LND01.xls

amh-mw committed:
[FIX] account_banking_ach_credit_transfer: fix tests

The due_date field in the account.payment.line.create wizard was
renamed to filter_date upstream in bank-payment/account_payment_order.

macagua committed:
Added translation using Weblate (Spanish)

macagua committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (32 of 32 strings)

Translation: l10n-usa-17.0/l10n-usa-17.0-l10n_us_mis_financial_report
Translate-URL: https://translation.odoo-community.org/projects/l10n-usa-17-0/l10n-usa-17-0-l10n_us_mis_financial_report/es/

Commit:
[MIG] l10n_us_mis_financial_report: Migration to 18.0

OCA-git-bot committed:
Merge PR #155 into 18.0

Signed-off-by jelenapoblet

OCA-git-bot committed:
Merge PR #162 into 18.0

Signed-off-by jelenapoblet

Commit:
[UPD] Update l10n_us_mis_financial_report.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #154 into 18.0

Signed-off-by jelenapoblet

Commit:
[UPD] Update l10n_us_base_county.pot

Commit:
[UPD] Update l10n_us_crm_county.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #141 into 18.0

Signed-off-by jelenapoblet

Commit:
[UPD] Update l10n_us_gaap.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #142 into 18.0

Signed-off-by jelenapoblet

Commit:
[UPD] Update l10n_us_gaap_mis_report.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #149 into 18.0

Signed-off-by jelenapoblet

Commit:
[UPD] Update account_banking_ach_base.pot

OCA-git-bot committed:
[BOT] post-merge updates

solomonprabu committed:
[ADD] Added Copyright info

OCA-git-bot committed:
Merge PR #171 into 18.0

Signed-off-by jelenapoblet

Commit:
[UPD] Update l10n_us_form_1099.pot

OCA-git-bot committed:
[BOT] post-merge updates
